clippy_lints/src/escape.rs

```diff
@@ -68,6 +68,11 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for BoxedLocal {
             if let ItemKind::Impl(_, _, _, _, Some(..), _, _) = item.kind {
                 return;
             }
+
+            // Issue 4804 fix: don't warn if the method is a default trait impl
+            if let ItemKind::Trait(_, _, _, _, _) = item.kind {
```

Comment:
We don't want to bail out for trait definitions in general. 

```
trait Trait {
    fn foo(x: Box<u32>) {}
}
```

should still get linted.

Reply:
Right, I forgot about associated methods in traits.
I'll try changing to code to skip only the methods with `self` in the arguments list.

Reply:
Changed in the new commit.